modules/controller/index.js

```diff
@@ -70,21 +70,16 @@ Controller.prototype.read = function (opts) {
     opts = {};
   }
   var source = this.source;
-  var type = source.typeName();
   var filters = this._filters.bind(this);
   var relations = this._relations.bind(this);
   // allow a custom responder to receive the payload
   var respond = opts.responder || responder;
   return function (request, response, next) {
-    var validRelations = relations(request).concat(opts.include || []);
     source.read({
       filters: filters(request),
-      relations: validRelations
+      relations: relations(request).concat(opts.include || [])
     }, function (err, data) {
-      var payload = readResponse(err, data, _.extend({}, opts, {
```

Comment:
Type isn't used at all in the read responder. It's passed in, but not used. https://github.com/endpoints/endpoints/blob/master/modules/controller/lib/responders/read.js

Do we anticipate needing it in the future? json-api spec says all primary data should be keyed under "data," not the type name

I don't think it should be in update, either, but left it for now.